core/Test/Tasty/Core.hs

```diff
@@ -408,23 +405,57 @@ foldTestTree
      -- ^ the tree to fold
   -> b
 foldTestTree (TreeFold fTest fGroup fResource fAfter) opts0 tree0 =
-  go mempty opts0 tree0
+  go (filterByPattern (evaluateOptions opts0 tree0))
+  where
+    go :: TestTreeWithOptions -> b
+    go = \case
+      EmptyTestTreeWO               -> mempty
+      SingleTestWO opts name test   -> fTest opts name test
+      TestGroupWO opts name trees   -> fGroup opts name (foldMap go trees)
+      WithResourceWO opts res0 tree -> fResource opts res0 $ \res -> go (tree res)
+      AfterWO opts deptype dep tree -> fAfter opts deptype dep (go tree)
+
+-- | 'TreeFold' is an eliminator for 'TestTreeWithOptions'.
```

Comment:
I'm wasn't familiar with the concept `eliminator`. Googling `eliminator haskell` gave 7 results pointing to the Hackage package `eliminators`, which doesn't go through any effort trying to explain what the concept is. More looking around only connects this term to dependent types which I don't think directly applies here.

In short; I think this comment needs to be reworded, preferably without use of very specific jargon.